cmd/clusterctl/pkg/client/cluster/move.go

```diff
@@ -202,11 +232,412 @@ func (o *objectGraph) Discovery(namespace string, types []metav1.TypeMeta) error
+		if len(nameSplit) != 2 {
+			continue
+		}
+
+		// If the secret is linked to a cluster by the naming convention, then add the secret to the list of the cluster's softDependents.
+		for _, cluster := range clusters {
+			if nameSplit[0] == cluster.identity.Name {
+				cluster.addSoftDependent(secret)
+			}
+		}
+	}
+}
+
+// Move moves all the Cluster API objects existing in a namespace (or from all the namespaces if empty) to a target management cluster
+func (o *objectGraph) Move() error {
+	clusters := o.getClusters()
+	o.log.Info("Clusters to move", "Count", len(o.uidToNode))
+
+	// Move Cluster by Cluster, ensuring all the dependent objects are moved as well and that the OwnerReferences relations are rebuilt in the target cluster.
+	// Nb. We are moving clusters one by one in sequential order for the sake of readability of the logs.
+	for _, cluster := range clusters {
+		o.log.Info("Moving", "Cluster", cluster.identity.Name, "Namespace", cluster.identity.Namespace)
+
+		// Sets the pause field on the Cluster object in the source management cluster, so the controllers stop reconciling it.
+		if err := setClusterPause(o.fromProxy, cluster, o.log); err != nil {
+			return err
+		}
+
+		// Create the Cluster object and all the dependent object tree into the target management cluster
+		// NB. we are doing all he create first, so the process can be restarted from the beginning time in case of problems.
+		if err := o.createClusterIntoTarget(cluster); err != nil {
+			return err
+		}
+
+		// Delete the Cluster object and all the dependent object tree from the source management cluster
+		// NB. we are doing delete after all the create completes, so the process can be restarted from the beginning time in case of problems.
+		if err := o.deleteClusterFromSource(cluster); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// setClusterPause sets the paused field on a Cluster object.
+func setClusterPause(proxy Proxy, cluster *node, log logr.Logger) error {
+	log.V(1).Info("Set Cluster.Spec.Paused", "Cluster", cluster.identity.Name, "Namespace", cluster.identity.Namespace)
+
+	cFrom, err := proxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	clusterObj := &clusterv1.Cluster{}
+	clusterObjKey := client.ObjectKey{
+		Namespace: cluster.identity.Namespace,
+		Name:      cluster.identity.Name,
+	}
+
+	if err := cFrom.Get(ctx, clusterObjKey, clusterObj); err != nil {
+		return errors.Wrapf(err, "error reading %q %s/%s",
+			clusterObj.GroupVersionKind(), clusterObj.GetNamespace(), clusterObj.GetName())
+	}
+
+	pausedClusterObj := clusterObj.DeepCopy()
+	pausedClusterObj.Spec.Paused = true
+
+	resourcePatch := client.MergeFrom(clusterObj)
+	if err := cFrom.Patch(ctx, pausedClusterObj, resourcePatch); err != nil {
+		return errors.Wrapf(err, "error pausing reconciliation for %q %s/%s",
+			pausedClusterObj.GroupVersionKind(), pausedClusterObj.GetNamespace(), pausedClusterObj.GetName())
+	}
+	return nil
+}
+
+var noOwner *node = nil
```

Comment:
Let's either move this declaration into `createClusterIntoTarget` where it's used, or just omit it and pass in `nil` directly.